docs/usage_guide/optimization_guide/index.rst

```diff
@@ -0,0 +1,213 @@
+  is 32, the column axis will not be split unless the amount of columns is greater than 32. If it is is greater, for example, 34, 
+  then the column axis is sliced into two partitions: containing 32 and 2 columns accordingly.
+
+Beware that ``NPartitions`` specifies a limit for the number of partitions `along a single axis`, which means, that
+the actual limit for the entire dataframe itself is the square of ``NPartitions``.
+
+.. figure:: /img/partitioning_mechanism/partitioning_examples.svg
+   :align: center
+
+Full-axis functions
+-------------------
+
+Some of the aggregation functions require knowledge about the entire axis, for example at ``.apply(foo, axis=0)``
+the passed function ``foo`` expects to receive data for the whole column at once.
+
+When a full-axis function is applied, the partitions along this axis are collected at a single worker
+that processes the function. After the function is done, the data is split again across all of the workers.
+
+.. figure:: /img/partitioning_mechanism/full_axis_function.svg
+   :align: center
+
+Note that the amount of remote calls is equal to the number of partitions, which means that since the number
+of partitions is decreased for full-axis functions it also decreases the potential for parallelism.
+
+Also note, that reduce functions such as ``.sum()``, ``.mean()``, ``.max()``, etc are not considered
+to be full-axis, so they do not suffer from the decreasing level of parallelism.
+
+How to tune partitioning
+------------------------
+
+As you can see from the examples above, the more the dataframe's shape is closer to a square, the closer the number of
+partitions to the square of ``NPartitions``. In the case of ``NPartitions`` equals to the number of workers,
+that means that a single worker is going to process multiple partitions at once, which slows down overall performance.
+
+If your workflow mainly operates with wide dataframes and non-full-axis functions, it makes sense to reduce the
+``NPartitions`` value so a single worker would process a single partition.
+
+.. figure:: /img/partitioning_mechanism/repartition_square_frames.svg
+   :align: center
+
+Copy-pastable example, showing how tuning ``NPartition`` value for wide frames may improve performance on your machine:
+
+.. code-block:: python
+
+  from multiprocessing import cpu_count
+  import modin.config as cfg
+  import modin.pandas as pd
+  import numpy as np
+  import timeit
+
+  # Generating data for a square-like dataframe
+  data = np.random.randint(0, 100, size=(2**10, 2**10))
+
+  # Explicitly setting `NPartitions` to its default value
+  cfg.NPartitions.put(cpu_count())
+
+  # Each worker process `sqrt(cpu_count())` amount of partitions
+  df = pd.DataFrame(data)
+  print(f"10 times of .abs(): {timeit.timeit(lambda: df.abs(), number=10)}s.")
+  print(f"NPartitions: {cfg.NPartitions.get()}")
+  # Possible output:
+  #     10 times of .abs(): 2.59s.
+  #     NPartitions: 112
```

Comment:
As I see, number of cores on you machine is 112. Data shape is (1024, 1024). For each axis we have 32 splits only. I think, more representable difference will be occurred when data shape will be increased to include 112 splits along both axis.
This will demonstrate case of `n_cores**2` total number of partitions.